pytorch_toolkit/formula_recognition/im2latex/models/backbones/resnet.py

```diff
@@ -0,0 +1,101 @@
+"""
+ Copyright (c) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import torch.nn as nn
+import torchvision.models
+
+architectures = {
+    "resnet18": torchvision.models.resnet18,
+    "resnet34": torchvision.models.resnet34,
+    "resnet50": torchvision.models.resnet50,
+    "resnet101": torchvision.models.resnet101,
+    "resnet152": torchvision.models.resnet152,
+    "resnext50_32x4d": torchvision.models.resnext50_32x4d,
+    "resnext101_32x8d": torchvision.models.resnext101_32x8d,
+}
+
+
+class ResNetLikeBackbone(nn.Module):
+    def __init__(self, configuration):
+        super(ResNetLikeBackbone, self).__init__()
+        disable_layer_3 = configuration.get('disable_layer_3')
+        disable_layer_4 = configuration.get('disable_layer_4')
+        arch = configuration.get('arch')
+        in_lstm_ch = configuration.get('in_lstm_ch', 512)
+        enable_last_conv = configuration.get('enable_last_conv', False)
+        self.arch = arch
+        if in_lstm_ch is not None:
+            self.in_lstm_ch = in_lstm_ch
+        else:
+            self.in_lstm_ch = 64
+        _resnet = architectures.get(arch, None)(
```

Comment:
You could've handled unknown backbones better.

Reply:
Default backbone is set to resnet50

Reply:
Is resnet50 an unknown backbone?